pkg/vetter/conflictingvirtualservicehost/vet.go

```diff
@@ -291,7 +292,15 @@ func addConflictsForSameRoute(trie *routeTrie, conflictingRules [][]routeRule) [
 	routeRules := trie.routeRules
 	for i := 0; i < len(routeRules)-1; i++ {
 		for j := i + 1; j < len(routeRules); j++ {
-			conflictingRules = append(conflictingRules, []routeRule{routeRules[i], routeRules[j]})
+			// Do not report when the rules are in the same namespace.
+			// Order matters for conflicting rules in the same namespace,
+			// however, this can be finnicky enough that I'm leaving it out
+			// of the first pass and we can add it in later if it is a cause
+			// of confusion.
```

Comment:
Agreed!

Reply:
Since this could cause confusion, I erred on the side of overstating this in a few places and to let someone digging through the code that this omission was intentional.

My idea for possibly rectifying this is to add a "precedence" field to the "routeRule" type, which can be passed through by the index innermost for loop in `buildMergedVirtualServiceTrie` and pass that down through `getRouteRuleFromMatch`. Then, in the conflict function, check if the rules still conflict given the precedence of the two route rules.